src/AudioUnit/AudioComponent.cs

```diff
@@ -38,6 +38,162 @@
+
+		public ResourceUsageInfo (NSDictionary dic) : base (dic) {}
+
+		public string[] IOKitUserClient { 
+			get {
+				var array = GetNativeValue<NSArray> (userClientK);
+				return NSArray.StringArrayFromHandle (array.ClassHandle);
+			} 
+			set {
+				SetArrayValue (userClientK, value);
+			}
+		}
+
+		public string[] MachLookUpGlobalName { 
+			get {
+				var array = GetNativeValue<NSArray> (globalNameK);
+				return NSArray.StringArrayFromHandle (array.ClassHandle);
+			} 
+			set {
+				SetArrayValue (globalNameK, value);
+			}
+		}
+
+		public bool? NetworkClient { 
+			get {
+				return GetBoolValue (networkClientK);
+			} 
+			set {
+				SetBooleanValue (networkClientK, value);
+			}
+		}
+
+		public bool? TemporaryExceptionReadWrite { 
+			get {
+				return GetBoolValue (exceptionK);
+			} 
+			set {
+				SetBooleanValue (exceptionK, value);
+			}
+		}
+	}
+
+	// keys are not constants and had to be found in AudioToolbox.framework/Headers/AudioComponent.h
+	[NoWatch, NoTV, Mac (10,13), iOS (11,0)]
+	public partial class AudioComponentInfo : DictionaryContainer {
+		static NSString typeK = new NSString ("type");
+		static NSString subtypeK = new NSString ("subtype");
+		static NSString manufacturerK = new NSString ("manufacturer");
+		static NSString nameK = new NSString ("name");
+		static NSString versionK = new NSString ("version");
+		static NSString factoryFunctionK = new NSString ("factoryFunction");
+		static NSString sandboxSafeK = new NSString ("sandboxSafe");
+		static NSString resourceUsageK = new NSString ("resourceUsage");
+		static NSString tagsK = new NSString ("tags");
+
+		public AudioComponentInfo () : base () {}
+
+		public AudioComponentInfo (NSDictionary dic) : base (dic) {}
+
+		public string Type { 
+			get {
+				return GetStringValue (typeK);
+			} 
+			set {
+				SetStringValue (typeK, value);
+			}
+		}
+
+		public string Subtype { 
+			get {
+				return GetStringValue (subtypeK);
+			} 
+			set {
+				SetStringValue (subtypeK, value);
+			}
+		}
+
+		public string Manufacturer { 
+			get {
+				return GetStringValue (manufacturerK);
+			} 
+			set {
+				SetStringValue (manufacturerK, value);
+			}
+		}
+
+		public string Name { 
+			get {
+				return GetStringValue (nameK);
+			} 
+			set {
+				SetStringValue (nameK, value);
+			}
+		}
+
+		public nuint? Version { 
+			get {
+				return GetNUIntValue (versionK);
+			} 
+			set {
+				SetNumberValue (versionK, value);
+			}
+		}
+
+		public string FactoryFunction { 
+			get {
+				return GetStringValue (factoryFunctionK);
+			} 
+			set {
+				SetStringValue (factoryFunctionK, value);
+			}
+		}
+
+		public bool? SandboxSafe { 
+			get {
+				return GetBoolValue (sandboxSafeK);
+			} 
+			set {
+				SetBooleanValue (sandboxSafeK, value);
+			}
+		}
+
+		public ResourceUsageInfo ResourceUsage { 
+			get {
+				return GetStrongDictionary<ResourceUsageInfo> (resourceUsageK);
+			} 
+			set {
+				SetNativeValue (resourceUsageK, value.Dictionary, true);
```

Comment:
`value.Dictionary` -> `NullReferenceException`
if `null` is allowed then you should use `value?.Dictionary`
if not then throw an exception